lspAst.sml

```diff
@@ -1,42 +1,113 @@
 structure lspAST =
 struct
-	datatype id = ID of string;
-	type message = string;
-	datatype method = CODECOMPELTION;
-	datatype params = CancelParams;
-	
-    datatype RequestMessage = RequestMessage of id * method * params;
-    
-    
-    datatype result = Result;
-    datatype errorCode = ParseError | InvalidRequest | MethodNotFound | InvalidParams
-                    | InternalError | ServerErrorStart | ServerErrorEnd
-                    | ServerNotInitialized | UnknownErrorCode | RequestCancelled;
-    type data = string;
-    datatype responseError = ResponseError of errorCode * message * data;
-    datatype error = Error of responseError;
-    datatype ResponseMessage = ResponseMessage of id * result * error;
-    
-    datatype cancelParams = CancelParams of id;
-    type scheme = string;
-    type authority = string;
-    type path = string;
-    type query = string;
-    type fragment = string;
-    datatype uri = DocumentUri of scheme * authority * path * query * fragment;
+	type Id = string;  (* Unique Id identifying a request/response *)
+	type Message = string;
+	datatype Method = CancelRequest | string;  (* Method called for the request *)
```

Comment:
The semi colons are not really required. It is only required in the interpreter. So get rid of the